File: 0346-moving-average-from-data-stream/0346-moving-average-from-data-stream.js
/**
 * @param {number} size
 */
var MovingAverage = function(size) {
    this.total = 0;
    this.max = size;
    this.window = [];
};

/** 
 * @param {number} val
 * @return {number}
 */
MovingAverage.prototype.next = function(val) {
    this.total += val;
    this.window.push(val);
    if (this.window.length > this.max) {
        this.total -= this.window.shift();
    }
    return this.total / this.window.length;
};

/** 
 * Your MovingAverage object will be instantiated and called as such:
 * var obj = new MovingAverage(size)
 * var param_1 = obj.next(val)
 */

File: 0346-moving-average-from-data-stream/0346-moving-average-from-data-stream.py
class MovingAverage:

    def __init__(self, size: int):
        self.total = 0
        self.maxLength = size
        self.window = []

    def next(self, val: int) -> float:
        self.total += val
        self.window.append(val)
        if len(self.window) > self.maxLength:
            self.total -= self.window.pop(0)
        return self.total / len(self.window)



# Your MovingAverage object will be instantiated and called as such:
# obj = MovingAverage(size)
# param_1 = obj.next(val)